components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -0,0 +1,219 @@
+package org.wso2.carbon.apimgt.impl.dao;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dao.constants.SQLConstants;
+import org.wso2.carbon.apimgt.impl.utils.GatewayArtifactsMgtDBUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GatewayArtifactsMgtDAO {
+    private static final Log log = LogFactory.getLog(GatewayArtifactsMgtDAO.class);
+    private static GatewayArtifactsMgtDAO INSTANCE = null;
+
+    /**
+     * Method to get the instance of the GatewayArtifactsMgtDAO.
+     *
+     * @return {@link GatewayArtifactsMgtDAO} instance
+     */
+    public static GatewayArtifactsMgtDAO getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new GatewayArtifactsMgtDAO();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Add details of the APIs published in the Gateway
+     *
+     * @param APIId        - UUID of the API
+     * @param APIName      - Name of the API
+     * @param version      - Version of the API
+     * @param tenantDomain - Tenant domain of the API
+     * @throws APIManagementException if an error occurs
+     */
+    public void addGatewayPublishedAPIDetails(String APIId, String APIName, String version, String tenantDomain)
+            throws APIManagementException {
+
+        try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
+                PreparedStatement statement = connection.prepareStatement(SQLConstants.ADD_GW_PUBLISHED_API_DETAILS)) {
```

Comment:
need to make autocommit false first in connection since it s an insert

Reply:
https://github.com/wso2/carbon-apimgt/pull/8828/commits/d501401001c713712bb50c42b40aa2afa9326545